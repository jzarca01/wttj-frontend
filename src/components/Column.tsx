import { useDroppable } from "@dnd-kit/core";

import { Card } from "./Card";

import { Column as ColumnType, Task } from "../types";

type Props = {
  tasks: Task[];
  column: ColumnType;
};

export const Column = ({ column, tasks }: Props) => {
  const { setNodeRef } = useDroppable({
    id: column.id,
  });

  return (
    <div className="flex w-80 flex-col rounded-lg bg-neutral-800 p-4">
      <h2 className="mb-4 font-semibold text-neutral-100">{column.title}</h2>
      <div ref={setNodeRef} className="flex flex-1 flex-col gap-4">
        {tasks.map((task) => {
          return <Card key={task.id} task={task} />;
        })}
      </div>
    </div>
  );
};
