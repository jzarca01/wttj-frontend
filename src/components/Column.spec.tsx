import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Column } from "./Column";

import type { Task, Column as ColumnType } from "../types";

describe("Column", () => {
  const mockTasks: Task[] = [
    { id: 1, title: "Task 1", description: "Description 1", status: "TODO" },
    { id: 2, title: "Task 2", description: "Description 2", status: "TODO" },
  ];

  const mockColumn: ColumnType = {
    title: "To Do",
    id: "TODO",
  };

  it("renders column title correctly", () => {
    render(<Column column={mockColumn} tasks={mockTasks} />);
    expect(screen.getByText("To Do")).toBeInTheDocument();
  });

  it("renders all tasks passed as props", () => {
    render(<Column column={mockColumn} tasks={mockTasks} />);

    expect(screen.getByText("Task 1")).toBeInTheDocument();
    expect(screen.getByText("Task 2")).toBeInTheDocument();
    expect(screen.getByText("Description 1")).toBeInTheDocument();
    expect(screen.getByText("Description 2")).toBeInTheDocument();
  });

  it("renders correctly with empty tasks array", () => {
    render(<Column column={mockColumn} tasks={[]} />);

    expect(screen.getByText("To Do")).toBeInTheDocument();
    expect(screen.queryByRole("heading", { level: 3 })).not.toBeInTheDocument();
  });
});
