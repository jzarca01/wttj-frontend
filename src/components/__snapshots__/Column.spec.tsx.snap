// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Column > renders column title correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-80 flex-col rounded-lg bg-neutral-800 p-4"
      >
        <h2
          class="mb-4 font-semibold text-neutral-100"
        >
          To Do
        </h2>
        <div
          class="flex flex-1 flex-col gap-4"
        >
          <div
            aria-describedby=""
            aria-disabled="false"
            aria-roledescription="sortable"
            class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
            role="button"
            tabindex="0"
          >
            <h3
              class="font-medium text-neutral-100"
            >
              Task 1
            </h3>
            <p
              class="mt-2 text-sm text-neutral-400"
            >
              Description 1
            </p>
          </div>
          <div
            aria-describedby=""
            aria-disabled="false"
            aria-roledescription="sortable"
            class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
            role="button"
            tabindex="0"
          >
            <h3
              class="font-medium text-neutral-100"
            >
              Task 2
            </h3>
            <p
              class="mt-2 text-sm text-neutral-400"
            >
              Description 2
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-80 flex-col rounded-lg bg-neutral-800 p-4"
    >
      <h2
        class="mb-4 font-semibold text-neutral-100"
      >
        To Do
      </h2>
      <div
        class="flex flex-1 flex-col gap-4"
      >
        <div
          aria-describedby=""
          aria-disabled="false"
          aria-roledescription="sortable"
          class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
          role="button"
          tabindex="0"
        >
          <h3
            class="font-medium text-neutral-100"
          >
            Task 1
          </h3>
          <p
            class="mt-2 text-sm text-neutral-400"
          >
            Description 1
          </p>
        </div>
        <div
          aria-describedby=""
          aria-disabled="false"
          aria-roledescription="sortable"
          class="cursor-grab rounded-lg bg-neutral-700 p-4 shadow-sm hover:shadow-md"
          role="button"
          tabindex="0"
        >
          <h3
            class="font-medium text-neutral-100"
          >
            Task 2
          </h3>
          <p
            class="mt-2 text-sm text-neutral-400"
          >
            Description 2
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
